File: Java/Program/Mahasiswa.java
public class Mahasiswa extends SivitasAkademik {
  public String nim;
  public String nama;
  public String gender;
  public String prodi;
  public String fakultas;

  public Mahasiswa() {}

  public Mahasiswa(String nik, String nama, String gender, String asal_universitas, String email_edu, String nim, String prodi, String fakultas) {
    super(nik, nama, gender, asal_universitas, email_edu);
    this.nim      = nim;
    this.prodi    = prodi;
    this.fakultas = fakultas;
  }

  public void setNim(String nim)
  {
    this.nim = nim;
  }

  public String getNim()
  {
    return this.nim;
  }

  public void setProdi(String prodi)
  {
    this.prodi = prodi;
  }

  public String getProdi()
  {
    return this.prodi;
  }

  public void setFakultas(String fakultas)
  {
    this.fakultas = fakultas;
  }

  public String getFakultas()
  {
    return this.fakultas;
  }

  public void displayData() {
    super.displayData();
    System.out.println("| NIM              : " + this.nim);
    System.out.println("| Program Studi    : " + this.prodi);
    System.out.println("| Fakultas         : " + this.fakultas);
  }
}

File: Python/Program/Mahasiswa.py
from SivitasAkademik import SivitasAkademik


class Mahasiswa(SivitasAkademik):

    __nim = ""
    __prodi = ""
    __fakultas = ""

    def __init__(self, nik, nama, gender, asal_universitas, email_edu, nim, prodi, fakultas):
        super().__init__(nik, nama, gender, asal_universitas, email_edu)
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas

    def set_nim(self, nim):
        self.__nim = nim

    def get_nim(self):
        return self.__nim

    def set_prodi(self, prodi):
        self.__prodi = prodi

    def get_prodi(self):
        return self.__prodi

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def get_fakultas(self):
        return self.__fakultas

    def display_data(self):
        super().display_data()
        print("| NIM              : ", self.__nim)
        print("| Program Studi    : ", self.__prodi)
        print("| Fakultas         : ", self.__fakultas)
